app/src/main/java/com/example/kharrison/my_application/CalculatorController.kt

```diff
@@ -0,0 +1,162 @@
+package com.example.kharrison.my_application
+
+import android.view.View
+import com.example.kharrison.my_application.CalculatorApp.Companion.ADDITION
+import com.example.kharrison.my_application.CalculatorApp.Companion.DIVISION
+import com.example.kharrison.my_application.CalculatorApp.Companion.MULTIPLICATION
+import com.example.kharrison.my_application.CalculatorApp.Companion.SUBTRACTION
+import kotlinx.android.synthetic.main.activity_calculator.view.*
+import kotlinx.android.synthetic.main.number_pad_layout.view.*
+import kotlinx.android.synthetic.main.opperations_layout.view.*
+
+
+class CalculatorController(val view : View) {
+
+   private var inputString = ""
+   private var outputString = ""
+
+   private var isOperationSelected = false
+   private var selectedOperator = 0
+
+   private var leftHandSide = ""
+   private var rightHandSide = ""
+   private var result = 0
+
+   private val mathHelper : CalcFunctions by lazy {
+      CalcFunctions()
+   }
+
+   init {
+      with(view) {
+         one.setOnClickListener { clickNumber("1") }
+         two.setOnClickListener { clickNumber("2") }
+         three.setOnClickListener { clickNumber("3") }
+         four.setOnClickListener { clickNumber("4") }
+         five.setOnClickListener { clickNumber("5") }
+         six.setOnClickListener { clickNumber("6") }
+         seven.setOnClickListener { clickNumber("7") }
+         eight.setOnClickListener { clickNumber("8") }
+         nine.setOnClickListener { clickNumber("9") }
+         zero.setOnClickListener { clickNumber("0") }
+         add.setOnClickListener { clickOperator(ADDITION) }
+         subtract.setOnClickListener { clickOperator(SUBTRACTION) }
+         multiply.setOnClickListener { clickOperator(MULTIPLICATION) }
+         divide.setOnClickListener { clickOperator(DIVISION) }
+         equals.setOnClickListener { calculate() }
+         clear.setOnClickListener { clearInput() }
+      }
+      invalidateControls()
+   }
+
+   private fun invalidateControls() {
+      val isEnabled = (!leftHandSide.isEmpty() || result != 0) && !isOperationSelected
+
+      view.add.isClickable = isEnabled
+      view.subtract.isClickable = isEnabled
+      view.divide.isClickable = isEnabled
+      view.multiply.isClickable = isEnabled
+   }
+
+   private fun updateText() {
+      view.input.text = inputString
+      view.answer.text = outputString
+   }
+
+   private fun updateOutPutString() {
+      leftHandSide = ""
+      rightHandSide = ""
+      inputString = ""
+      updateText()
+   }
+
+   private fun addToInputString(input : String) {
+      inputString += input
+      updateText()
+   }
+
+   private fun clickNumber(num : String) {
+      if (!isOperationSelected) {
+         leftHandSide += num
+      } else {
+         rightHandSide += num
+      }
+      addToInputString(num)
+      invalidateControls()
+   }
+
+   private fun clickOperator(operator : Int) {
+      isOperationSelected = true
+      when (operator) {
```

Comment:
You might consider just doing a static map that maps between the enum and the symbol. Then you could something like....

    selectedOperator = operator
    addToInputString( operatorMap[operator] )

Or you could make use real enums for `ADDITION`, etc and have them define the strings. Then you could something like...

    selectedOperator = operator
    addToInputString( operator.symbol )